boards: arm: update nRF5340 documentation to point to the nRF5340 DK (#29917)
zephyrproject-rtos/zephyr: boards/arm/nrf5340dk_nrf5340/doc/index.rst

> * a secondary ARM Cortex-M33 core, with a reduced feature set, running at

Reviewer: ```suggestion
* a secondary Arm Cortex-M33 core, with a reduced feature set, running at
```
Global change.

Author: updated.

>   a fixed 64 MHz, referred to as the **Network MCU**.

Reviewer: ```suggestion
  a fixed 64 MHz, referred to as the **network core**.
```
Same change for Application MCU -> application core, and both throughout the document.

Author: updated now, @ru-fu

> The nrf5340dk_nrf5340_cpuapp board target provides support for the Application
> MCU on nRF5340 SoC. The nrf5340dk_nrf5340_cpunet board target provides support
> for the Network MCU on nRF5340 SoC.

Reviewer: ```suggestion
The nrf5340dk_nrf5340_cpuapp build target provides support for the application
core on the nRF5340 SoC. The nrf5340dk_nrf5340_cpunet build target provides support
for the network core on the nRF5340 SoC.
```

Author: updated now, @ru-fu

.. figure:: img/nrf5340pdk.png
     :width: 711px
     :align: center

>      :alt: nRF5340 DK

Reviewer 2: The image itself should also be updated.

Author: updated :)

| WDT       | on-chip    | watchdog             |
+-----------+------------+----------------------+

The nrf5340dk_nrf5340_cpunet board configuration supports the following
hardware features:

> +-----------+------------+----------------------+

Reviewer 2: The table below should be corrected. ADC and PWM are not available in the Network core.

Author: Thanks, not sure why this was wrong anyways,
Updated now.

> +-----------+------------+----------------------+

Reviewer: Line 146/147 should be nRF5340 DK (not Development Kit)

Author: Yeap, updated all ARM to Arm.

Programming and Debugging
*************************

> nRF5340 application core supports the Armv8m Security Extension.

Reviewer: ```suggestion
nRF5340 application core supports the Armv8-M Security Extension.
```

Applications build for the nrf5340dk_nrf5340_cpuapp board by default
boot in the Secure state.

> nRF5340 network core does not support the Armv8m Security Extension.

Reviewer: ```suggestion
nRF5340 network core does not support the Armv8-M Security Extension.
```

Author: updated. This is now consistent, at least within the nRF5340 Docs.
In general we need to see how to spell ARM :)

*************************

> nRF5340 Application MCU supports the Armv8m Security Extension.

Reviewer: ```suggestion
The nRF5340 application core supports the Armv8-M Security Extension.
```
Global change.

Author: updated

Applications build for the nrf5340dk_nrf5340_cpuapp board by default
boot in the Secure state.

> nRF5340 Network MCU does not support the Armv8m Security Extension.

Reviewer: ```suggestion
The nRF5340 network core does not support the Armv8-M Security Extension.
```

.. _IDAU:
   https://developer.arm.com/docs/100690/latest/attribution-units-sau-and-idau

> .. _nRF5340 DK website:

Author: @ru-fu i wonder if I should just remove this link completely and re-add it once the DK web page is up..

Reviewer: I would keep it, because it will most likely redirect to the DK website.
